### ISSUE TITLE
[Docs]: Mention createGraphQLHandler will be handled later

### ISSUE THREAD

**Issue by razzeee** [redwoodjs/redwood#7558]
### Summary and description

https://github.com/redwoodjs/redwood/blob/main/docs/docs/auth/custom.md?plain=1#L196 mentions that `createGraphQLHandler` will be handled later, but it's never spoken of again.

### Are you interested in working on this?

- [ ] I'm interested in working on this

**Comment by razzeee** [comment]
Seems like using an `authDecoder` is now mandatory, even if it's not needed for the flow. Otherwise I'm running into user errors.

Here's what I'm doing for now.
```
export const authDecoder = async (token: string, type: string) => {
  if (type !== 'custom-auth') {
    return null
  }

  return ''
}
```

then use that in `graphql.ts`:

```

export const handler = createGraphQLHandler({
  getCurrentUser,
  authDecoder,
  loggerConfig: { logger, options: {} },
...
```

**Comment by jtoar** [comment]
@razzeee you're right, that part is missing, and I'll get on fixing that. But I'm curious what kind of authentication flow doesn't need a decoder? Is it not JWT-based?

**Comment by razzeee** [comment]
Not actually sure it qualifies as auth in the traditional sense, but I'm basically getting a token from a gitlab instance/app. I guess I could decode that, but I need to call the user endpoint on gitlab anyway as soon as somebody logs in, so gitlab will block anything without a valid token.

**Comment by jtoar** [comment]
@razzeee gotcha, let me get my head around the flow again, but it sounds like a solution here is just to make `authDecoder` optional? It sounds like you'd just rather not decode the token since you have no use for it on the api side?

**Comment by razzeee** [comment]
Yeah, then again, it shouldn't hurt to have that empty implementation above, it was just confusing, as it didn't work as before I guess. 

It might also be a bit more secure, if I check it in that function in addition. If I can do that, I'll need to check, but I'll already left the office for this week.

**Comment by jtoar** [comment]
@razzeee no worries, and we wouldn't want to have to force you to pass an empty implementation, so definitely down to figure this out. Ran this by @Tobbe and he said that authDecoder isn't required:  

https://github.com/redwoodjs/redwood/blob/c6a31f1da3329538ad94cd5a1847bbb780162898/packages/api/src/auth/index.ts#L66-L70

So we may need to dig a little deeper here to see what's going on. Next week when you're back could you elaborate on the user error a bit? Thank!

**Comment by razzeee** [comment]
My header basically crashes, as there is no user it can figure out, but I end up in the user code of the header. Which is kinda intended, as there's only a landing page your supposed to be on without login.

It didn't seem to call getCurrentUser ( in the old implementation that would be here https://github.com/razzeee/redwood-gitlab-auth/commit/4bf527c5ab203e22c6ab0a743f3190bb0e96f009#diff-d1122ca2901576c7820b7efaddd0e32dbc8b14d888c11d89e820048736f4c006R28 )

**Comment by jtoar** [comment]
@razzeee I see, since authDecoder returns undefined, the whole block here where we call `getCurrentUser` is skipped:

https://github.com/redwoodjs/redwood/blob/5782a18773b208034321c023b1d7acfd434317f6/packages/graphql-server/src/plugins/useRedwoodAuthContext.ts#L42-L54

**Comment by jtoar** [comment]
@razzeee thanks for bringing this to our attention and debugging with us, Tobbe and I will think about a way forward here. It looks like you already have one—just make an `authDecoder` function that does something. But we'll see if we can improve the flow for you. That would be ideal right? If you could just provide a `getCurrentUser` function that gets called?

**Comment by razzeee** [comment]
Either that, or some kind of mention in the custom auth docs I guess. I probably only ran into this due to migrating our old implementation.

But yeah, it's confusing, that there is an invisible dependency between `getCurrentUser` and `authDecoder` while `createGraphQLHandler` doesn't enforce both to be set.

**Comment by jtoar** [comment]
@razzeee just wanted you to know that https://github.com/redwoodjs/redwood/pull/7572 is in progress to provide a legit path towards an optional authDecoder

**Comment by Tobbe** [comment]
And it's merged and available in canary 🙂

**Comment by jtoar** [comment]
@razzeee this has been released in v4.1.1: https://github.com/redwoodjs/redwood/releases/tag/v4.1.1. If this fixes it for you I think we can close this one out but please do let us know!

**Comment by razzeee** [comment]
Looking good in dev and testing, will need some more testing, before I can push it out, but should be fine.
